python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,215 @@
+        return obj
+
+    @classmethod
+    def read_tuple(self, stream, tuplesize):
+        datalist = []
+        for i in range(tuplesize):
+            # first element read type
+            element_type = self.read_int(stream)
+            # Second read the length
+            length = self.read_int(stream)
+
+            if length == SpecialLengths.NULL or length == 0:
+                datalist.append(0)
+            # length is > 0, read the item now
+            elif length > 0:
+                obj = self.read_item(stream, element_type, length)
+                datalist.append(obj)
+
+        return datalist
+
+    @classmethod
+    def write_with_length( self, obj, stream, output_type):
+        if output_type == DataType.INT:
+            self.write_int(DataType.INT, stream)
+            self.write_int(obj, stream)
+        elif output_type == DataType.LONG:
+            self.write_int(DataType.LONG, stream)
+            self.write_long(obj, stream)
+        elif output_type == DataType.FLOAT:
+            self.write_int(DataType.FLOAT, stream)
+            self.write_float(obj, stream)
+        elif output_type == DataType.DOUBLE:
+            self.write_int(DataType.DOUBLE, stream)
+            self.write_double(obj, stream)
+        elif output_type == DataType.BLOB:
+            self.write_int(DataType.BLOB, stream)
+            self.pickle_and_write(self, obj, stream)
+
+    @classmethod
+    def read_command(self, stream):
+        length = self.read_int(stream)
+
+        if length < 0:
+            raise ValueError("Command cannot be less than zero.")
+        s = stream.read(length)
+
+        if len(s) < length:
+            raise EOFError
+        unenc = base64.urlsafe_b64decode(s)
+        return self.loads(unenc)
+
+    @staticmethod
+    def dumps(obj):
+        protocol = 2
+        return cPickle.dumps(obj, protocol)
+
+    @staticmethod
+    def loads(obj):
+        return cPickle.loads(obj)
+
+    @staticmethod
+    def pickle_and_write(self, obj, stream):
+        serialized = self.dumps( obj)
+
+        if serialized is None:
+            raise ValueError("Serialized value should not be None.")
+        elif len(serialized) > (1 << 31):
+            raise ValueError("Cannot serialize object larger than 2G.")
+
+        self.write_int(len(serialized), stream)
+        stream.write(serialized)
+
+
+def main(in_file, out_file):
+    pickle_serializer = PickleSerializer()
+    try:
+        func = pickle_serializer.read_command(in_file)
+        tuple_size = pickle_serializer.read_int(in_file)
+        output_type = pickle_serializer.read_int(in_file)
+        is_flatmap = pickle_serializer.read_int(in_file)
+
+        if tuple_size < 1:
+            raise ValueError("size of tuple should not be less than 1 ")
+
+        while True:
+            num_tuples = pickle_serializer.read_int(in_file)
+            if num_tuples == SpecialLengths.END_OF_STREAM:
+                exit(0)
```

Comment:
If the effect of this `exit()` is exactly the same as a `break`, then you should use `break` (for the same reason you don't use `goto` if you can avoid it: it obscures control flow).